src/xunit.performance.api/PerformanceTestMessageSink.cs

```diff
@@ -78,5 +87,15 @@ private static Type GetType(string assemblyName, string typeName)
 
             return null;
         }
+
+        public override void Dispose()
+        {
+            Discovery.DiscoveryCompleteMessageEvent -= OnDiscoveryCompleteMessageEvent;
+            Discovery.TestCaseDiscoveryMessageEvent -= OnTestCaseDiscoveryMessageEvent;
+
+            base.Dispose();
+        }
+
+        public ManualResetEvent Finished => _finished;
```

Comment:
Could you please remove the private member variable `_finished`, and use property instead (like the `Test` property above):
```csharp
...
public PerformanceTestMessageSink()
{
    Finished = new ManualResetEvent(false);
    Tests = new List<PerformanceTestMessage>();
...
public ManualResetEvent Finished { get; }
```